scripts/send_data_to_serial.py

```diff
@@ -0,0 +1,72 @@
+from tendo import singleton
+
+# script lock
+me = singleton.SingleInstance()
+
+import json
+import logging
+import os
+import subprocess
+import sys
+from datetime import datetime
+from typing import List
+
+import serial
+from adafruit_board_toolkit import circuitpython_serial
+
+formatter = logging.Formatter(
+    fmt="%(asctime)s %(levelname)-8s %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
+)
+logger = logging.getLogger("serial")
+logger.setLevel(logging.DEBUG)
+fh = logging.FileHandler("serial.log")
+fh.setLevel(logging.DEBUG)
+fh.setFormatter(formatter)
+logger.addHandler(fh)
+
+# File needs to be in the same dir as this script
+powershell_script_name = "getActiveWindowProcessName.ps1"
+
+
+def current_active_window_process_name():
```

Comment:
The dependency on a PowerShell script is rather unideal and slows the whole process of refreshing the MacroPad down quite a bit (though the total time is quite low still). 

Since this script is presumably being called by some trigger, that trigger is likely aware of the currently active window name and could pass it as a parameter to the script. This would have the added benefit of making this script more os agnostic. 

With that said, I had significant trouble getting this to work with AutoHotKey.

Reply:
I wonder if something like this would work? https://stackoverflow.com/a/58355052/2899120